# Classifying Urban Sounds using Voting Classifier on ANN and CNN

In [1]:
%matplotlib inline
# from memory_profiler import memory_usage
import os
import pandas as pd
from glob import glob
import numpy as np
import os
import pandas as pd

In [2]:
# Read in the results from our Dense NN
preds_ann = pd.read_csv('predict_ann.csv')

In [3]:
# Read in the results from our CNN
preds_cnn = pd.read_csv('predict_cnn.csv')

In [4]:
# Convering the results to arrays 
preds_cnn_array = preds_cnn.to_numpy()
preds_ann_array = preds_ann.to_numpy()

In [5]:
# Adding the porcentages of predictions together
preds = (preds_cnn_array + preds_ann_array)

In [6]:
# Reading the original data to create our testing dataframe
data = pd.read_csv(r"C:\Users\רועי\data\Urban-Sounds\train\data.csv")

In [7]:
# Generating a new test dataframe that includes the labels for comparison and 
# checking the accuracy of our model with this never before seen data
test = data[4435:]

# Creating an empty list to store the values where the predictions are the maximum out 
# of all the 10 possible values
p = []
for i in range(0, len(preds)):
    p.append(np.where(preds[i] == max(preds[i])))

# Creating an empty list to store the values in a clean list
predictions = []
for i in range(0, len(preds)):
    predictions.append(p[i][0][0])
    
# Adding those predictions to our test dataframe
test['predictions'] = predictions

# Changing the numeric values to their corresponding labels
test['predictions'] = test['predictions'].map({8:'siren', 9:'street_music', 7:'jackhammer',
                   4:'drilling', 3:'dog_bark', 2:'children_playing',
                   6:'gun_shot', 5:'engine_idling', 0:'air_conditioner',
                   1:'car_horn'})

# Checking the percentage of correct predictions 
round(len(test[test['Class'] == test['predictions']])/len(test),2)

C:\ProgramData\Anaconda3\lib\site-packages\ipykernel_launcher.py:17: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
C:\ProgramData\Anaconda3\lib\site-packages\ipykernel_launcher.py:23: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy


0.94

### We get 94% accuracy by combining both the ANN and the CNN models!